app/sessionStore.js

```diff
@@ -823,29 +823,43 @@ module.exports.runPreMigrations = (data) => {
     delete data.sites
   }
 
-  if (data.lastAppVersion) {
-    try {
-      // with version 0.22.13, any file downloaded (including the update itself) would get
-      // quarantined on macOS (per work done with https://github.com/brave/muon/pull/484)
-      // this functionality was then reverted with https://github.com/brave/muon/pull/570
-      //
-      // To fix the executable, we need to manually un-quarantine the Brave executable so that it works as expected
-      if (process.platform === 'darwin' && compareVersions(data.lastAppVersion, '0.22.13') === 0) {
-        let runningAppPath
-        const appDirectories = app.getPath('exe').split(path.sep)
-        // Remove the `Contents`/`MacOS`/`Brave` parts from path
-        if (appDirectories.length > 3) {
-          runningAppPath = path.sep + path.join(...appDirectories.slice(0, appDirectories.length - 3))
+  if (data.lastAppVersion || data.quarantineNeeded) {
+    // with version 0.22.13, any file downloaded (including the update itself) would get
+    // quarantined on macOS (per work done with https://github.com/brave/muon/pull/484)
+    // this functionality was then reverted with https://github.com/brave/muon/pull/570
+    //
+    // To fix the executable, we need to manually un-quarantine the Brave executable so that it works as expected
+    if (process.platform === 'darwin' && (compareVersions(data.lastAppVersion, '0.22.13') === 0 || data.quarantineNeeded)) {
+      const unQuarantine = (appPath) => {
+        try {
+          execSync(`xattr -d com.apple.quarantine "${appPath}"`)
+          console.log(`Quarantine attribute has been removed from ${appPath}`)
+        } catch (e) {
+          console.error(`Failed to remove quarantine attribute from ${appPath}: `, e)
         }
+      }
+
+      console.log('Update was downloaded from 0.22.13' + data.quarantineNeeded ? ' (first launch after auto-update)' : '')
 
-        execSync('xattr -d com.apple.quarantine /Applications/Brave.app || true')
-        if (runningAppPath) {
-          execSync(`xattr -d com.apple.quarantine "${runningAppPath}" || true`)
+      // Un-quarantine default path
+      const defaultAppPath = '/Applications/Brave.app'
+      unQuarantine(defaultAppPath)
+
+      // Un-quarantine custom path
+      const appDirectories = app.getPath('exe').split(path.sep)
+      if (appDirectories && appDirectories.length > 3) {
+        // Remove the `Contents`/`MacOS`/`Brave` parts from path
+        const runningAppPath = path.sep + path.join(...appDirectories.slice(0, appDirectories.length - 3))
```

Comment:
This will remove the last 3 characters of the string.
Check: https://jsfiddle.net/pzg7k9v5/

Instead the code should be:
```
var splitAppDirectories = appDirectories.split(path.sep)
const runningAppPath = splitAppDirectories.slice(0, -3).join(path.sep))
```

See here: https://jsfiddle.net/wq44La9z/2/